src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -494,6 +494,9 @@ public static void SendCommand(EventSource eventSource, EventCommand command, ID
         /// </summary>
         public override string ToString()
         {
+            if (!IsSupported)
+                return string.Empty;
```

Comment:
What is the benefit of this change... being able to trim out SR.EventSource_ToString if EventSource isn't supported?

Reply:
Yes to trim the method body and references and easier identification of EventSource instances when event source support is removed

Reply:
> easier identification of EventSource instances when event source support is removed

Can you clarify this?  Should this instead be `return base.ToString()!;`?

Reply:
> Can you clarify this?

Both values it prints are uninitialized and I don't think it provides any value to print null value.

> Should this instead be return base.ToString()!;?

No strong preference but not sure it's worth dealing with the null.

Reply:
> Both values it prints are uninitialized and I don't think it provides any value to print null value

Why does object.ToString return null?

Reply:
> The actual base object.ToString method will never return null.

I see we bang it everywhere and just hide the null values.

Reply:
> and just hide the null values

We use ! with it because we know it'll never return null.